[release-3.6] Set dependency between slurmd and nvidia-persistenced

jdeamicis opened the pull request:
### Description of changes
* Configure systemd dependency between slurmd and nvidia-persistenced services in order to make sure NVIDIA device files are available under /dev before slurmd starts.

### Tests
* Add kitchen tests for the modified code to validate the systemd dependencies configured for the slurmd service.
* Manually ran the added kitchen tests on all EC2 platforms.
* Manually created a cluster with a patch to verify that the dependency is successfully configured on GPU nodes and not on non-GPU ones.

### References
* https://github.com/aws/aws-parallelcluster-cookbook/pull/2100

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [x] Check all commits' messages are clear, describing what and why vs how.
- [x] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [ ] ~Check if documentation is impacted by this change.~

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

jdeamicis committed:
Set dependency between slurmd and nvidia-persistenced

Configure systemd dependency between slurmd and nvidia-persistenced services
in order to make sure NVIDIA device files are available under /dev before
slurmd starts.

Add kitchen tests for the modified code.

Signed-off-by: Jacopo De Amicis <jdamicis@amazon.it>

NSsirena commented on test/recipes/controls/aws_parallelcluster_slurm/config_slurmd_systemd_service_spec.rb:
```diff
@@ -0,0 +1,38 @@
+  describe 'Check slurmd systemd "after" dependencies'
+  describe command('systemctl list-dependencies --after --plain slurmd.service') do
+    its('stdout') { should include "munge.service" }
+    its('stdout') { should include "munge.service" }
```
Duplicated line.
Do you expected another service to be present in the list of services ?

jdeamicis replied:
I will remove it in the next commit.

jdeamicis committed:
Change type of dependency between slurmd and nvidia-persistenced

Set systemd dependency to "Wants" to get a more robust system.
In case of problems with nvidia-persistenced, slurmd may still run
if the GPU device files are available.

This also prevents the dependency from potentially breaking GPU nodes
where nvidia-persistenced is not supported.

Signed-off-by: Jacopo De Amicis <jdamicis@amazon.it>
